src/main/java/hunternif/mc/atlas/network/client/TilesPacket.java

```diff
@@ -33,9 +31,15 @@ public TilesPacket(DimensionType dimension) {
 		tileData = Unpooled.buffer();
 	}
 
+	public TilesPacket(DimensionType dimension, int tileCount, ByteBuf tileData) {
+		this.dimension = dimension;
+		this.tileCount = tileCount;
+		this.tileData = tileData;
+	}
+
 	public TilesPacket addTile(int x, int y, int biomeID) {
-		tileData.writeShort(x);
-		tileData.writeShort(y);
+		tileData.writeInt(x);

```

Comment:
TODO: review my approach, if I should support interop between different versions of the mod. Probably not, so let's delete `IntTilesPacket`. Also make sure other fixes from 1.12 are merged in.